src/main.rs

```diff
@@ -224,32 +224,47 @@ fn handle_input_events(
         Err((Some(ENODEV), _)) => Ok(false),
         Err((_, error)) => Err(error).context("Error fetching input events"),
         Ok(events) => {
+            //A vector used to collect all mouse movement events
+            let mut mousemovements: Vec<RelativeEvent> = Vec::new();
+
             for event in events {
                 if let Some(event) = Event::new(event) {
-                    handle_event(handler, dispatcher, config, event)?;
+                    //Running the event through modmap and keymap
+                    let handled_events = handle_event(handler, config, event)?;
+                    //and checking if the resulting events are mouse movements
+                    for action in handled_events {
```

Comment:
I think you needed to aggregate mouse movement events here because the event handler accepts events one by one. However, as I explained before, the point of the current code structure is to make it testable by just examining the input and the output of the event handler. If you implement this logic outside it, then it's kind of nullified.

WDYT about changing `handle_event` / `on_event` to `handle_events` / `on_events`? Then I think you can handle this inside the event handler.